src/components/PaletteSwitcherDialog/index.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+  Logic for saving and updating the site color scheme.
+  Contains:
+  useWatchUsersColorScheme - method for setting up event listeners to detect changes to user color 
+  preferences
+  PaletteSwitcherDialog - component with a dialog box to allow users to choose their color
+  preferences in the site, and save their preference in localStorage
+*/
+
+import GordonDialogBox from 'components/GordonDialogBox';
+import React, { useEffect, useState } from 'react';
+import {
+  FormControl,
+  FormControlLabel,
+  FormLabel,
+  RadioGroup,
+  Radio,
+  Divider,
+} from '@mui/material';
+import { storageColorPreferenceKey } from 'theme';
+import { useColorScheme } from '@mui/material/styles';
+
+// Watches for system setting changes, and updates the color accordingly
+export const useWatchUsersColorScheme = () => {
+  const { setMode } = useColorScheme();
+
+  useEffect(() => {
+    const systemPrefersDark = window.matchMedia('(prefers-color-scheme: dark)');
+
+    const onSystemColorSchemeChange = (e: any) => {
+      const storedColorScheme = localStorage.getItem(storageColorPreferenceKey) ?? 'system';
+
+      if (storedColorScheme === 'system') {
+        setMode(e.matches ? 'dark' : 'light');
+      }
+    };
+
+    systemPrefersDark.addEventListener('change', onSystemColorSchemeChange);
+
+    return () => {
+      systemPrefersDark.removeEventListener('change', onSystemColorSchemeChange);
+    };
+  }, [setMode]);
+};
+
+type Props = {
+  handleClose: () => void;
+  dialogOpen: boolean;
+};
+
+const PaletteSwitcherDialog = ({ dialogOpen, handleClose }: Props) => {
+  // defaults to system setting if no user setting is stored
+  const [localScheme, setLocalScheme] = useState(
+    localStorage.getItem(storageColorPreferenceKey) ?? 'system',
+  );
+  const { setMode } = useColorScheme();
+
+  useEffect(() => {
+    SetModeFromSetting(localScheme);
+    /* ensures the color scheme is always correct, previously when the system setting was 
+    changed while the page was closed, the page would render still on the old scheme */
+  }, []);
+
+  // sets the site color scheme based on the setting string passed to it, defaults to light
+  const SetModeFromSetting = (scheme: string) => {
+    setMode(
+      scheme === 'system'
+        ? window.matchMedia('(prefers-color-scheme: dark)').matches
+          ? 'dark'
+          : 'light'
+        : scheme === 'dark'
+        ? 'dark'
+        : 'light',
```

Comment:
personal opinion. This is a spot where using ternary operators makes the code significantly more convoluted. having a switch case  or `if/else`  on `scheme` would be better